nixos/modules/programs/iio-niri.nix

```diff
@@ -0,0 +1,58 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    mkEnableOption
+    mkPackageOption
+    mkOption
+    types
+    mkIf
+    getExe
+    concatStringsSep
+    ;
+  cfg = config.services.iio-niri;
+in
+{
+  options.services.iio-niri = {
+    enable = mkEnableOption "IIO-Niri";
+
+    package = mkPackageOption pkgs "iio-niri" { };
+
+    niriUnit = mkOption {
+      type = types.nonEmptyStr;
+      default = "niri.service";
+      description = "The Niri **user** service unit to bind IIO-Niri's **user** service unit to.";
+    };
+
+    extraArgs = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      description = "Extra arguments to pass to IIO-Niri.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    hardware.sensor.iio.enable = true;
+
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.user.services.iio-niri = {
```

Comment:
This should probably add `iio-sensor-proxy.service` to `requires` and `after` as well.

Reply:
I see, good catch. I suppose we'll just leave it then, the module *does* enable the sensor unit, and the user unit restarts on failure, so I guess the case of a race condition is covered.